### PR TITLE
Final wrapup added documentation and output files for future reference

### PR DESCRIPTION
**Pull request by Puneet-Jain-18**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,71 @@
-# Cricket_Shot_Commentry
-Minor Project of College for V Semester
+# Automated_Cricket_Shot_Commentry
+The problem which we are addressing is about recognizing different shots played by a
+batsman in a cricket match by analysing the video of the played shot and providing
+commentary for the same.Our task was to minimize the latency in delivering the commentary and maximizing the success rate.Henceforth we decided to implement a deep
```
Comment (Vibhu-Agarwal) [inline README.md]: Typo: Missed a space between two sentences

File: README.md [hunk 2]
```diff
@@ -1,2 +1,71 @@
...
+frames and then applying CNN on it and the maximum confidence score of the shot which we achieve over all the frames, we classify shot to that category.
+
+# Processing The Input
+
+1. Video was broken into respective frames(images).
+2. For each frame obtained, the following steps were done.<br />
+  a. Conversion of images to grey scale.<br />
+  b. Changing dimension of image to 100x100.<br />
+  c.Necessary rotations of images were done(where ever required).<br />
+  d.Reducing intensity range from (0,255) to (0,1). <br /><br />
```
Comment (Vibhu-Agarwal) [inline README.md]: Inconsistent Formatting of alphabetical bullets and following spacing

File: README.md [hunk 3]
```diff
@@ -1,2 +1,71 @@
...
+
+All the data gathered from above steps were combined together to make a single
+csv file.
+![alt text](https://github.com/Puneet-Jain-18/Cricket_Shot_Commentry/blob/master/finalOutputs/Preprocessing.png)
+
+# Generation Of Commentry
+For video and audio integration, few audio clips were generated from a app, approximately 5-7 sample audio clips per shot. The video samples were cropped to get 3-4 sec video clip from youtube, it was done so as to get maximum frames of batsman while playing the shot.For deployment of our service we first pick up frames from video consecutively and feed that image to our CNN model. The model generates a csv file of the classified shot probability which it gets in each frame. After which another separate csv is generated which provides the max count value of the the shot in all frames and after getting the frame number in which the shot is correctly identified the audio is synchronized with the video in that frame.The audio file is chosen randomly from the available ones
```
Comment (Vibhu-Agarwal) [inline README.md]: Typo: Missed space between the second and the third sentence

File: README.md [hunk 4]
```diff
@@ -1,2 +1,71 @@
...
+
+# CNN Model Features
+
+At first we provide a 100x100 pixel grayscale image to our model after which, In our model we have<br />
+A. 3, convolutional layers of 64,128 and 236 of 3x3 filter, each of which have Relu as activation<br />
+B. 3 Max pooling layers of size 2x2 each applied after convolutional layer.<br />
+C. 3 Dropout layers of with 0.25 parameter.<br />
+D. 1 Flatten layer<br />
+E. 2 Dense layers in which for one Relu activation function is used and in the other<br />
+F. one softmax is used to get probability of each class<br />
```
Comment (Vibhu-Agarwal) [inline README.md]: Typo: Missed a capitalization of the first letter of the sentence

File: README.md [hunk 5]
```diff
@@ -1,2 +1,71 @@
...
+
+![alt text](https://github.com/Puneet-Jain-18/Cricket_Shot_Commentry/blob/master/finalOutputs/CNN.png)
+
+# Running The Project
```
Comment (Vibhu-Agarwal) [inline README.md]: Replace with **`Usage`**

File: README.md [hunk 6]
```diff
@@ -1,2 +1,71 @@
...
+The project is based on linux OS and we have used python and R as major languages for development.
+To build the project 
+1. Open terminal and locate to video_processing directory.Now Run commands
```
Comment (Vibhu-Agarwal) [inline README.md]: Add a bash command code snippet (`cd pathto/video_processing`)

File: README.md [hunk 7]
```diff
@@ -1,2 +1,71 @@
...
+````
+source venv/bin/activate
+pip3 install -r requirements.txt
+pip freeze > requirements.txt
```
Comment (Vibhu-Agarwal) [inline README.md]: Unnecessary command

File: README.md [hunk 8]
```diff
@@ -1,2 +1,71 @@
...
+F. one softmax is used to get probability of each class<br />
+
+![alt text](https://github.com/Puneet-Jain-18/Cricket_Shot_Commentry/blob/master/finalOutputs/CNN.png)
+
+# Running The Project
+The project is based on linux OS and we have used python and R as major languages for development.
+To build the project 
+1. Open terminal and locate to video_processing directory.Now Run commands
+````
+source venv/bin/activate
```
Comment (Vibhu-Agarwal) [inline README.md]: Creation of virtual environment needed before activating it.
Provide commands for creating one, if you have to enforce virtual environments.

Otherwise, drop the need for creating virtual environments altogether.

File: README.md [hunk 9]
```diff
@@ -1,2 +1,71 @@
...
+pip3 install -r requirements.txt
+pip freeze > requirements.txt
+````
+This should install the required python dependencies in your system.
+
+2. Now to install R dependencies run script r_dependencies
+````
+Rscirpt "install_dependencies.R"
+````
+Ensure that all the packages have been sucessfully installed.If any of thoes fails you would need to build it from source form the corresponding github repo.
```
Comment (Vibhu-Agarwal) [inline README.md]: Typo: `thoes`

File: README.md [hunk 10]
```diff
@@ -1,2 +1,71 @@
...
+3. Now we need to pick out frames from the videos for CNN to process them.
```
Comment (Vibhu-Agarwal) [inline README.md]: Bullet merged with previous sentence in markdown

File: README.md [hunk 11]
```diff
@@ -1,2 +1,71 @@
...
+````
+./final_shell_script.sh
+````
+this command would extract frame from videos, preprocess them for different filters and give them to the CNN model for classification. CNN model shall return the probability of each frame for each class. ie we would have a matrix of (No. of frames)X6. Now we need to make decision on which shot is being played in the video on basis of that information.
+4. Our final script makes this decision on basis of differnt parameters like no of shots classified in particular class and max confidence level for a frame etc. Run
```
Comment (Vibhu-Agarwal) [inline README.md]: Bullet merged with the previous sentence in markdown

File: README.md [hunk 12]
```diff
@@ -1,2 +1,71 @@
...
+````
+./decision_shell.sh
+````
+This shall make our decision and then integrate commentry to video iteslf such that the audio of video is replaced with the actural commentry for that video.<br/>
+<b>The video is placed in final_output directory.</b><br />
+
+![alt text](https://github.com/Puneet-Jain-18/Cricket_Shot_Commentry/blob/master/finalOutputs/sample_output.png)
+# Future Work
+This was a really fun project and helped us learn a lot about deep learning and neural nets. We were able to obtain high accuracy for image classification but could not obtain such high accuracy on video classification.
+1. One looking for extending this project could make use of pretrained models like vgg or resnet50 to improve accuracy of videos and overcome the boundation of limited data we have.
```
Comment (Vibhu-Agarwal) [inline README.md]: Incorrect usage of words: `One looking for ...`

File: README.md [hunk 13]
```diff
@@ -1,2 +1,71 @@
...
+2. the dataset iteslf could be reconstructed as current dataset only shows the batsman whereas in videos we have a total field view containing bowler, umpire etc which is affection the accuracy of model.
+3. 3-D CNN could be directly applied if one has sufficient amount of video dataset. This could extremely increase the robustness of model and lead to a better classification.
+<h4>Thanks-For-Your-Time</h4>
```
Comment (Vibhu-Agarwal) [inline README.md]: Convert from HTML to Markdown
